app/tutorials/traffic-light-communication/src/main/java/org/eclipse/mosaic/app/tutorial/TrafficLightApp.java

```diff
@@ -65,13 +67,19 @@ private void setRed() {
     public void onMessageReceived(ReceivedV2xMessage receivedV2xMessage) {
         if (receivedV2xMessage.getMessage() instanceof GreenWaveMsg) {
             getLog().infoSimTime(this, "Received GreenWaveMsg");
-            if (((GreenWaveMsg) receivedV2xMessage.getMessage()).getMessage().equals(SECRET)) {
-                getLog().infoSimTime(this, "Received correct passphrase: {}", SECRET);
-
-                if (DEFAULT_PROGRAM.equals(getOs().getCurrentProgram().getProgramId())) {
-                    setGreen();
+            //Assure that distanceTo does not throw a NullPointerException
+            assert receivedV2xMessage.getMessage().getRouting().getSource().getSourcePosition() != null;
```

Comment:
Code starting with the `assert` keyword are skipped when executed with java by default, and are only executed in tests or using the jvm-parameter `-ea`. Instead, we use the following line, which still throws a NullPointerException, but with a error message:
```java
Validate.notNull(receivedV2xMessage.getMessage().getRouting().getSource().getSourcePosition(), "The source position of the sender cannot be null");
```